source/slang/slang-ir-diff-jvp.cpp

```diff
@@ -1401,30 +1446,25 @@ struct JVPTranscriber
 
     InstPair transcribeIfElse(IRBuilder* builder, IRIfElse* origIfElse)
     {
-        // The loop comes with three blocks.. we just need to transcribe each one
-        // and assemble the new loop instruction.
+        // IfElse Statements come with 4 blocks. We transcribe each block into it's
+        // linear form, and then wire them up in the same way as the original if-else
         
-        // Transcribe the target block (this is the 'condition' part of the loop, which
-        // will branch into the loop body).
-        // Note that for the condition we use the primal inst (condition values should not have a 
-        // differential)
+        // Transcribe condition block
         auto primalConditionBlock = findOrTranscribePrimalInst(builder, origIfElse->getCondition());
         SLANG_ASSERT(primalConditionBlock);
 
-        // Transcribe the break block (this is the block after the exiting the loop)
+        // Transcribe 'true' block (condition block branches into this if true)
         auto diffTrueBlock = findOrTranscribeDiffInst(builder, origIfElse->getTrueBlock());
         SLANG_ASSERT(diffTrueBlock);
 
-        // Transcribe the continue block (this is the 'update' part of the loop, which will
-        // branch into the condition block)
+        // Transcribe 'false' block (condition block branches into this if true)
+        // TODO (sai): What happens if there's no false block?
```

Comment:
Should just wrap the false block into an `if` and don't transcribe it if it is null.